### ISSUE TITLE
Discussion: Refactor checks

### ISSUE THREAD

**Issue by carslen** [eclipse-tractusx/tractusx-quality-checks#13]
# What

Refactor checks / repository.

# Why

Primarily for reasons of clarity.

## Items To Refactor
* `internal` Directory 
  The more TRG checks are implemented, the more confusing the `internal` directory will look like. Later may also other checks be added. Group them into packages named like our TRG categories (TRG 1, TRG 2, etc.)

* create reusable generic tests, instead of duplicating checks (e.g. file Checks in TRG 1 and TRG 2)
* rethink package, type, func naming
  * `txqualitychecks` is too general and contains currently runners, checks, ...
  * `txqualitychecks.NewTestRunner()` vs. `txqualitychecks.TestRunner.New()`
  * `ChangeLogExists`(struct) vs. `NewChangelogExists` (func). func name is misleading.
  * ...

**Comment by SebastianBezold** [comment]
- internal: Yes we can introduce packages here. As of now, i did not see the need, since the codebase is still small and there are no real differentiators between the checks, that make package naming obvious
- reusable parts: definitely yes. As of now, we could move the "file exists" part out. Not yet done, since it's 4 LoC and moving + calling the function also creates more code
- package names: yes definitely rename. Any suggestions? My ideas so far:
  - Some of the release guidelines that are basis for the checks should not only be tested on release but always -> that why I went with quality. 
  - TX, to indicate, that it is specific for Eclipse Tractus-X and is most likely not adaptable to other projects
  - `NewTestRunner` -> I think this is a well known pattern in golang to create instances of structs. This is used to implement proper defaults. One could argue, that we don't need that at all for the current structs, since most of them do not have any properties. I used it, to have the same method of initializing things and a caller does not have to figure out how to do it. -> Reduce number of surprises
